examples/gallery/images/grdgradient.py

```diff
@@ -0,0 +1,59 @@
+"""
+Calculating the gradient and radiance of a grid file
+----------------------------------------------------
+The :meth:`pygmt.grdgradient` method calculates the gradient of a grid file.
+In the example shown below we will see how to calculate a hillshade map based
+on a Data Elevation Model (DEM). As input :meth:`pygmt.grdgradient` gets
+a :class:`xarray.DataArray` object or a path string to a grid file. We will use
+the ``radiance`` parameter in order to set the illumination source direction
+and altitude.
+"""
+
+import pygmt
+
+fig = pygmt.Figure()
+
+# Define region of interest around Yosemite valley
+region = [-119.825, -119.4, 37.6, 37.825]
+
+# Load sample grid (3 arc second global relief) in target area
+grid = pygmt.datasets.load_earth_relief(resolution="03s", region=region)
```

Comment:
The example takes ~11 seconds to execute because the 03s earth relief grid is not fully cached in the GitHub Actions. Currently, only the tile in [135, 136, 35, 36] is cached.

We have two solutions:

1. Change the `region` parameter to `[135, 136, 35, 36]` to use the cached 03s earth relief grid tile
2. Cache the tile for region `[-119.825, -119.4, 37.6, 37.825]`

Reply:
In my opinion, it is nice to give an example of a famous place like Yosemite valley, is it problematic to add those cached files?

Reply:
The tile size is about 1.5 MB. I think it's OK to add it to the cache.

To add the tile to the cache, you need to:

1. Add `N37W120.earth_relief_03s_g.nc` (this is the grid file name for this tile) after this line: https://github.com/GenericMappingTools/pygmt/blob/a99533ff30cffaaf5b6da84244ba5d773276c256/pygmt/helpers/testing.py#L163
2. Uncomment the "pull_request" line in the [cache_data.yml](https://github.com/GenericMappingTools/pygmt/blob/a99533ff30cffaaf5b6da84244ba5d773276c256/.github/workflows/cache_data.yaml#L6) workflow
https://github.com/GenericMappingTools/pygmt/blob/a99533ff30cffaaf5b6da84244ba5d773276c256/.github/workflows/cache_data.yaml#L6
3. commit the changes, so that the workflow will run in this PR to refresh the caches
4. after the workflow finishes, the cached data should contain the tile used in this example. Comment the "pull_request" line in the `cache_data.yml` workflow.